src/EnergyPlus/HVACInterfaceManager.cc

```diff
@@ -691,12 +691,18 @@ namespace HVACInterfaceManager {
 
         } else { // tank has mass
             if (MassFlowRate > 0.0) {
-                TankFinalTemp = (LastTankOutletTemp - (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp)) *
-                                    std::exp(-(MassFlowRate * Cp) / (ThisTankMass * Cp) * TimeStepSeconds) +
+                // original code is: std::exp(-(MassFlowRate * Cp) / (ThisTankMass * Cp) * TimeStepSeconds)
+                Real64 stdexpMCp = (MassFlowRate * Cp) / (ThisTankMass * Cp) * TimeStepSeconds;
+                if (stdexpMCp < 700.0) {
```

Comment:
Adding protection here is fine.

Reply:
Someone else also made a change here to protect divide by 0. I'm just going with that other change to pull this out of this PR.